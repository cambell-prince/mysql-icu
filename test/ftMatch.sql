-- Test full text parsing
set names utf8;
create database if not exists collate_my_MM;

use collate_my_MM;

drop table if exists matchTest;
drop table if exists matchTest2;

create table matchTest  (
    id INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
    phrase VARCHAR(128)
) collate utf8_icu_my_MM_ci;
create table matchTest2  (
    id INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
    phrase VARCHAR(128)
) collate utf8_icu_custom_ci;

create fulltext index icuft on matchTest (phrase) with parser icu;
create fulltext index icuft2 on matchTest2 (phrase) with parser icu;

insert into matchTest2 (phrase) values 
('မြန်မာစကား'),
('စကာလုံးများ'),
('ဘယ်နှလုံးလဲ။'),
('ဘာလုပ်မလဲ။'),
('မြန်မာဘာသာစကား'),
('မြန်မာလူမျို'),
('မြန်မာနိုင်ငံ'),
('နိုင်ငံရေး'),
('လူများတယ်'),
('အင်္ဂလိန်ကမြန်မာပြည်ကိုလားတယ်။'),
('မြန်လားမာလား။');

insert into matchTest2 (phrase) values 
('This is English'),
('It is easier to use full text searches in English');

insert into matchTest select * from matchTest2;

repair table matchTest quick;
repair table matchTest2 quick;

select 'လုံး';
select * from matchTest where match (phrase) against ('လုံး');
select 'လုပ်';
select * from matchTest where match (phrase) against ('လုပ်');
select 'မြန်မာစကား';
select * from matchTest where match (phrase) against ('မြန်မာစကား');
select 'မြန်မာ';
select * from matchTest where match (phrase) against ('မြန်မာ');
select 'မြန်မာ*';
select * from matchTest where match (phrase) against ('မြန်မာ*' IN BOOLEAN MODE);
select 'လူ';
select * from matchTest where match (phrase) against ('လူ');
select '+လူ';
select * from matchTest where match (phrase) against ('+လူ' IN BOOLEAN MODE);

select 'Custom rules';
select 'လုံး';
select * from matchTest2 where match (phrase) against ('လုံး');
select '+လူ';
select * from matchTest2 where match (phrase) against ('+လူ' IN BOOLEAN MODE);

select '+လူ -"မြန်မာ"';
select * from matchTest2 where match (phrase) against ('+လူ -မြန်' IN BOOLEAN MODE);


select 'မြန်မာ';
select * from matchTest2 where match (phrase) against ('မြန်မာ');
select '"မြန်မာ" boolean';
select * from matchTest2 where match (phrase) against ('"မြန်မာ"' IN BOOLEAN MODE);
select 'မြန်မာ*';
select * from matchTest2 where match (phrase) against ('မြန်မာ*' IN BOOLEAN MODE);

