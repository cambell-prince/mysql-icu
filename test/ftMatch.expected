Table	Op	Msg_type	Msg_text
mysql_icu_test.matchTest	repair	status	OK
Table	Op	Msg_type	Msg_text
mysql_icu_test.matchTest2	repair	status	OK
Match လုံး
Match လုံး
Match လုပ်
Match လုပ်
Match မြန်မာစကား
Match မြန်မာစကား
id	phrase
1	မြန်မာစကား
Match မြန်မာ
Match မြန်မာ
Match မြန်မာ*
Match မြန်မာ*
id	phrase
1	မြန်မာစကား
7	မြန်မာနိုင်ငံ
5	မြန်မာဘာသာစကား
6	မြန်မာလူမျို
Match လူ
Match လူ
Match +လူ
Match +လူ
Custom rules
Custom rules
Match လုံး
Match လုံး
id	phrase
2	စကာလုံးများ
3	ဘယ်နှလုံးလဲ။
Match +လူ
Match +လူ
id	phrase
6	မြန်မာလူမျို
9	လူများတယ်
Match +လူ -မြန်
Match +လူ -မြန်
id	phrase
9	လူများတယ်
Match မြန်မာ
Match မြန်မာ
id	phrase
1	မြန်မာစကား
6	မြန်မာလူမျို
7	မြန်မာနိုင်ငံ
5	မြန်မာဘာသာစကား
10	အင်္ဂလိန်ကမြန်မာပြည်ကိုလားတယ်။
11	မြန်လားမာလား။
Match "မြန်မာ" boolean
Match "မြန်မာ" boolean
id	phrase
1	မြန်မာစကား
5	မြန်မာဘာသာစကား
6	မြန်မာလူမျို
7	မြန်မာနိုင်ငံ
10	အင်္ဂလိန်ကမြန်မာပြည်ကိုလားတယ်။
Match မြန်မာ*
Match မြန်မာ*
id	phrase
1	မြန်မာစကား
5	မြန်မာဘာသာစကား
6	မြန်မာလူမျို
7	မြန်မာနိုင်ငံ
10	အင်္ဂလိန်ကမြန်မာပြည်ကိုလားတယ်။
11	မြန်လားမာလား။
Match ကော
Match ကော
id	phrase
12	စကောဘာသာစကား
Match ကောင်း
Match ကောင်း
id	phrase
13	ကောင်းတယ်။
